src/Shared/Microsoft.DotNet.Kusto/KustoHelpers.cs

```diff
@@ -207,17 +224,59 @@ public void Dispose()
                 return _inner.IngestFromStreamAsync(stream, ingestionProperties, sourceOptions);
             }
         }
+
+        private class NonDisposableCslAdmin : ICslAdminProvider
+        {
+            private readonly ICslAdminProvider _inner;
+            public string DefaultDatabaseName { get; set; }
+
+            public NonDisposableCslAdmin(ICslAdminProvider inner, string defaultDatabaseName)
+            {
+                _inner = inner;
+                DefaultDatabaseName = defaultDatabaseName;
+            }
+
+            public void Dispose()
+            {
+                // This is non-disposable
+            }
+
+            public IDataReader ExecuteControlCommand(
+                string databaseName, 
+                string command,
+                ClientRequestProperties properties = null)
+            {
+                return _inner.ExecuteControlCommand(databaseName, command, properties);
+            }
+
+            public Task<IDataReader> ExecuteControlCommandAsync(
+                string databaseName, 
+                string command,
+                ClientRequestProperties properties = null)
+            {
+                return _inner.ExecuteControlCommandAsync(databaseName, command, properties);
+            }
+
+            public IDataReader ExecuteControlCommand(
+                string command,
+                ClientRequestProperties properties = null)
+            {
+                return _inner.ExecuteControlCommand(command, properties);
+            }
+        }
     }
 
     public interface IKustoIngestClientFactory
     {
         IKustoIngestClient GetClient();
+        ICslAdminProvider GetAdminProvider();
     }
 
     public class KustoOptions
     {
         public string QueryConnectionString { get; set; }
         public string IngestConnectionString { get; set; }
+        public string AdminConnectionString { get; set; }
```

Comment:
Hrm... I wonder if we need another connection string here.  It will always, presumably, be the same as one of the others (we only have 1 user account, after all, "nethelix").  I think the "ingest" string is different because ingestion is fundamentally different than querying (you can't use the same connection string for both), but we should, theoretically, be able to reuse the "QueryConnectionString", but that's... annoying. :-)

Reply:
the QueryConnectionString can be used for the admin connection, I just created a separated one to make the use of this more intentional. I think that sometimes you can use the "QueryConnectionString" without worry about breaking something (especially when you are testing or running something locally/development) because it is just for queries and if we use it for the admin operations too, I think it can lead to people running in a performing action as the append or deletes without them wanting or be really aware of the admin powers of this secret